### PR TITLE
document which lifetime is needed for systemparam derive

### PR DESCRIPTION
**Pull request by laundmo**
# Objective

Document a few common cases of which lifetime is required when using SystemParam Derive

## Solution

Added a table in the doc comment

### CODE REVIEW

**Review by alice-i-cecile — COMMENTED** [review]
Great idea to document, but I think this will be a lot more robust as a doc test.

**Comment by Nilirad** [comment]
> Great idea to document, but I think this will be a lot more robust as a doc test.

I agree, actual doc examples are more useful and less prone to desync. Also, the user can obtain the lifetime parameters from the specific `SystemParam` they're interested in.

**Comment by laundmo** [comment]
Hope this is somewhat like you imagined? didn't want to make it visibly a codeblock, as there already is a example there. sadly no way to preserve table formatting during this... alas.

**Review by alice-i-cecile — APPROVED** [review]

**Review by Nilirad — APPROVED** [review]
Looks good enough. Not comprehensive, but sufficiently covers the most important cases. Ultimately, lifetimes can still be seen on the struct docs of the specific `SystemParam`.

**Review by james7132 — APPROVED** [review]

**Review by JoJoJet — APPROVED** [review]

File: crates/bevy_ecs/src/system/system_param.rs [hunk 1]
```diff
@@ -36,6 +36,35 @@ use std::{
 /// Derived `SystemParam` structs may have two lifetimes: `'w` for data stored in the [`World`],
 /// and `'s` for data stored in the parameter's state.
 ///
+/// The following list shows the most common SystemParams and which lifetime they require
```
Comment (alice-i-cecile) [inline crates/bevy_ecs/src/system/system_param.rs:39]:
```suggestion
/// The following list shows the most common [`SystemParam`]s and which lifetime they require
```

**Pull request merged**